apps/crn-server/src/routes/research-outputs.route.ts

```diff
@@ -27,11 +30,35 @@ export const researchOutputRouteFactory = (
   researchOutputRoutes.get(
     '/research-outputs',
     async (req, res: Response<ListResearchOutputResponse>) => {
-      const { query } = req;
+      const { query, loggedInUser } = req;
+      const { associationId, status, ...options } =
+        validateResearchOutputFetchOptions(query);
+
+      const isRequestingDrafts = status === 'draft';
+
+      const hasTeamRole =
+        getUserRole(loggedInUser as UserResponse, 'teams', [associationId]) !==
+        'None';
 
-      const options = validateFetchOptions(query);
+      const hasWorkingGroupRole =
+        getUserRole(loggedInUser as UserResponse, 'workingGroups', [
+          associationId,
+        ]) !== 'None';
+
+      if (isRequestingDrafts && !hasTeamRole && !hasWorkingGroupRole) {
+        throw Boom.forbidden();
+      }
 
-      const result = await researchOutputController.fetch(options);
+      const result = await researchOutputController.fetch({
+        ...options,
+        ...(isRequestingDrafts && {
```

Comment:
i dont know why you keep insisting on handling the case individually when drafts are requested and then use a different set of filters? 
Previously we had the `includeDrafts` option right? That option allowed us to ALSO return drafts, yeah? So that should stay.

and now we are introducing a status filter - where we want to say 'give me either published or unpublished outputs', correct? To me you could just leave them separate and if they do `includeDrafts: false` and `status: Draft` (or whatever the value is for the status) then return an empty list. 

As for the `associationId` - sorry but i still don't fully understand - in FE do we know what is being provided, so `workingGroupId` or `teamId`? I'm pretty sure we do in which case you should have a filter for those two and not a generic `associationId`, especially if we introduce more filters like this.